configs/components/yaml-cpp.rb

```diff
@@ -23,9 +23,14 @@
   elsif platform.is_macos?
     cmake_toolchain_file = ""
     cmake = "/usr/local/bin/cmake"
-    if platform.is_cross_compiled?
+    if platform.architecture == 'arm64'
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos11' if platform.name =~ /osx-11/
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos12' if platform.name =~ /osx-12/
+      pkg.environment 'CXX', 'clang++ -target arm64-apple-macos13' if platform.name =~ /osx-13/
+
+      if platform.os_version.to_i >= 13
+        cmake = "/opt/homebrew/bin/cmake"
+      end
```

Comment:
I left a comment about this in https://github.com/puppetlabs/puppet-runtime/pull/680/files#r1253955838